cedar-policy-core/src/entities/conformance.rs

```diff
@@ -176,6 +179,66 @@ pub fn typecheck_value_against_schematype(
     }
 }
 
+/// Check whether the given `RestrictedExpr` is a valid instance of `SchemaType`
+pub fn does_restricted_expr_implement_schematype(
+    expr: BorrowedRestrictedExpr<'_>,
+    expected_ty: &SchemaType,
+) -> bool {
+    use SchemaType::*;
+
+    match expected_ty {
+        Bool => expr.as_bool().is_some(),
+        Long => expr.as_long().is_some(),
+        String => expr.as_string().is_some(),
+        EmptySet => expr.as_set_elements().is_some(),
+        // PANIC SAFETY: LHS checks that `is_some()`.
+        #[allow(clippy::unwrap_used)]
+        Set { .. }
+            if expr.as_set_elements().is_some() && expr.as_set_elements().unwrap().count() == 0 =>
```

Comment:
might be able to avoid the unwrap with something like
```
Set { .. } if matches!(expr.as_set_elements(), Some(iter) if iter.count() == 0) =>
```

Reply:
That gives the error:
```
cannot move out of `iter` in pattern guard
variables bound in patterns cannot be moved from until after the end of the pattern guard
```
and I didn't find a work around.